pkg/controller/deorbit/deorbiter.go

```diff
@@ -138,6 +170,20 @@ func (d *ConcreteDeorbiter) DeleteServices() (deleted []core_v1.Service, err err
 	return deleted, err
 }
 
+func (d *ConcreteDeorbiter) WaitForSnapshotCleanUp() (err error) {
+	done, err := d.isSnapshotCleanupFinished()
+
+	if err != nil {
+		return err
+	}
+
+	if done {
+		return nil
+	}
+
+	return wait.PollUntil(PollInterval, d.isSnapshotCleanupFinished, d.Stop)
```

Comment:
Can be replaced with `wait.PollImmediateUntil()`.

Reply:
done